src/include/sof/math/exp_fcn_hifi4.h

```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOF_MATH_EXPONENTIAL_HIFI4_H__
+#define __SOF_MATH_EXPONENTIAL_HIFI4_H__
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define EXPONENTIAL_GENERIC
+#if defined(__XCC__)
+
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI4 == 1
+#undef EXPONENTIAL_GENERIC
+#define EXPONENTIAL_HIFI4
+#endif
+
+#endif
+
+/* define constants */
+#define BIT_MASK_LOW_Q27P5 0x0000000008000000
```

Comment:
Will need a prefix for these since these are in public API header e.g.
```MATH_EXP_```
or move them to private header.
@singalsu what prefix are you using today for maths APIs/Macros ?

Reply:
Please accept my sincere apologies;  I am unaware of the headers for the public API.
If you could leave me a note with some instructions, a website, etc., I would be grateful.

Reply:
We don't have the MATH_ or math_ prefix in most of src/math and src/include/sof/math stuff. I think it would be good. @ShriramShastry could you do the math prefix add for you exp functions PRs. I can do a PR to address the existing functions similarly to get this cleaned up.

Reply:
Math_ could leave conflicts, so SOFM_ and sofm_ should be good. Can you add those Sriram?